test/wasm_test.cc

```diff
@@ -195,6 +196,13 @@ TEST_P(TestVm, AlwaysApplyCanary) {
             auto wasm_handle_comp =
                 createWasm(vm_key, source, plugin_comp, wasm_handle_factory_comp,
                            wasm_handle_clone_factory_for_canary, false);
+            // Validate that canarying is cached for the first baseline plugin variant.
+            if (first) {
+              first = false;
+              EXPECT_EQ(canary_count, 0);
+              continue;
+            }
+
             // For each create Wasm, canary should be done.
             EXPECT_EQ(canary_count, 1);
```

Comment:
Should this be `if/else` instead so that other tests are also performed on the first pass?

Reply:
First iteration is explicitly done as "baseline" above.

Reply:
What I meant is that because of added `continue` all the tests below are skipped for the first plugin variant, and those tests are designed to verify that you're getting the same Wasm and Plugin instances for that first and baseline plugin variants.

Reply:
Ok, updated to restore the previous coverage.